datasets/guardian_authorship/guardian_authorship.py

```diff
@@ -0,0 +1,314 @@
+                                                   2)),
+                           train_folder="Books", valid_folder="Society", test_folder="Politics,UK,World"),
+
+        GuardianAuthorshipConfig(name="cross_genre_{}".format(3),
+                           version=nlp.Version("{}.0.0".format(15),
+                                               description="The Original DS with the cross-genre scenario no.{}".format(
+                                                   3)),
+
+                           train_folder="Books", valid_folder="UK", test_folder="Politics,Society,World"),
+
+        GuardianAuthorshipConfig(name="cross_genre_{}".format(4),
+                           version=nlp.Version("{}.0.0".format(16),
+                                               description="The Original DS with the cross-genre scenario no.{}".format(
+                                                   4)),
+                           train_folder="Books", valid_folder="World", test_folder="Politics,Society,UK"),
+    ]
+
+    # @property
+    def _info(self):
+        # TODO: Specifies the nlp.DatasetInfo object
+        return nlp.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # nlp.features.FeatureConnectors
+            features=nlp.Features(
+                {
+                    # These are the features of your dataset like images, labels ...
+                    "author": nlp.features.ClassLabel(names=["catherinebennett", "georgemonbiot", "hugoyoung",
+                                                             "jonathanfreedland", "martinkettle", "maryriddell",
+                                                             "nickcohen", "peterpreston", "pollytoynbee",
+                                                             "royhattersley", "simonhoggart", "willhutton",
+                                                             "zoewilliams"]),
+                    "topic": nlp.features.ClassLabel(names=["Politics", "Society", "UK", "World", "Books"]),
+                    "article": nlp.Value("string"),
+                }
+            ),
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=[("article", "author")],
+            # Homepage of the dataset for documentation
+            homepage="http://www.icsd.aegean.gr/lecturers/stamatatos/papers/JLP2013.pdf",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: Downloads the data and defines the splits
+        # dl_manager is a nlp.download.DownloadManager that can be used to
+        # download and extract URLs
+        dl_dir = dl_manager.download_and_extract(_URL)
+        # print("-------------------------------")
```

Comment:
Could you remove/clean your comments ?

Reply:
Cleaned, and added a better description of how to use the dataset. I wasn't aware of the "draft pull request". Sorry for all the back and forth